backend/src/http.rs

```diff
@@ -100,40 +103,38 @@ impl Service<Request<Body>> for RootService {
                 Box::pin(result)
             }
 
-            (&Method::GET, path) if Assets::get(&path[1..]).is_some() => {
-                // TODO: don't call `Assets::get` twice.
-                Assets::serve(&path[1..])
+            // Realm pages
+            (&Method::GET, path) if path.starts_with(REALM_PREFIX) => {
+                let _realm_path = path.strip_prefix(REALM_PREFIX).unwrap();
```

Comment:
I know :(

I tried to improve that, but couldn't come up with a nice way how to. On nightly one can `(&Method::GET, path) if let Some(realm_path) = path.strip_prefix(REALM_PREFIX) => ...`. But that's unfortunately not stable yet and doesn't look like it will be for some time. I also thought about wrapping all that in a macro, but that wasn't that great either. So I just decided to go with this solution.

Although... I just noticed that I think I wanted to write `let _realm_path = &path[REALM_PREFIX.len()..];` instead. Yeah, will change that.

Reply:
Oh, that's in nightly already, cool! I only saw the RFC in my quick search.